external-crates/move/crates/move-unit-test/src/test_runner.rs

```diff
@@ -281,6 +294,11 @@ impl SharedTestingConfig {
                 err.remove_exec_state();
             }
         }
+        let trace = if self.trace_location.is_some() {
+            Some(move_tracer.into_trace())
+        } else {
+            None
+        };
```

Comment:
This may be too cute, but I couldn't resist.

```suggestion
        let trace = if self.trace_location
            .is_some()
            .then(|| move_tracer.into_trace());
```